Classes/Domain/Repository/DatabaseStorageRepository.php

```diff
@@ -67,4 +81,58 @@ public function findStorageidentifiers()
 
         return $this->identifiers;
     }
+
+    /**
+     * Delete all entries of a storage by its identifier and an optional date interval.
+     *
+     * @param string $storageIdentifier Storage identifier
+     * @param \DateInterval|null $dateInterval Date interval
+     * @return int
+     * @throws IllegalObjectTypeException
+     * @throws InvalidQueryException
+     */
+    public function deleteByStorageIdentifierAndDateInterval(string $storageIdentifier, \DateInterval $dateInterval = null): int
+    {
+        $query = $this->createQuery();
+        $constraints = [
+            $query->equals('storageidentifier', $storageIdentifier)
+        ];
+
+        if ($dateInterval !== null) {
+            $currentDate = new \DateTime();
+            $newDate = $currentDate->sub($dateInterval);
+            $constraints[] = $query->lessThan('datetime', $newDate->format('Y-m-d H:i:s'));
+        }
+
+        $query->matching($query->logicalAnd($constraints));
+        $entries = $query->execute();
+        $count = 0;
+        foreach ($entries as $entry) {
+            $this->remove($entry);
+            $count++;
```

Comment:
Removing resources should be quite simple here, sth. like:

```php
foreach ($entry->getProperties() as $property) {
    if (!$property instanceof PersistentResource) {
        // Skip non-resources
        continue;
    }
    
    $resourceManager->deleteResource($property);
}
```

But this could be sth. for a new PR, as removal of resources is missing in general right now.

Reply:
I had the same thought. Maybe this should be removed to the DatabaseStorageService class. But for now, that's fine for me.